src/random.cpp

```diff
@@ -36,9 +36,13 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "libtorrent/aux_/disable_warnings_push.hpp"
 
+#ifdef BOOST_NO_CXX11_HDR_RANDOM
```

Comment:
is this reliable? I would expect that perhaps you could do something like: ``#ifdef __has_include && __has_include(<random>)